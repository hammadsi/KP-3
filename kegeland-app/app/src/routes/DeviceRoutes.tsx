import {createStackNavigator} from '@react-navigation/stack';
import React from 'react';

import AppHeader from '~components/AppHeader';
import withDevice from '~hoc/withDevice';
import withQuestionnaire from '~hoc/withQuestionnaire';
import DevicesScreen from '~views/devices/DevicesScreen';
import FemfitScreen from '~views/devices/FemfitScreen';

import {DeviceStackParamList} from './interface';

const DeviceStack = createStackNavigator<DeviceStackParamList>();

/**
 * DeviceRoutes. Configure the routes within the device stack
 */
const DeviceRoutes: React.FC = () => {
  return (
    <DeviceStack.Navigator
      initialRouteName="Devices"
      screenOptions={{
        header: (props) => <AppHeader {...props} />,
        headerShown: true,
      }}>
      <DeviceStack.Screen name="Devices" component={DevicesScreen} />
      <DeviceStack.Screen
        name="Femfit"
        component={withQuestionnaire(
          'femfit',
          withDevice('femfit', FemfitScreen),
        )}
      />
    </DeviceStack.Navigator>
  );
};

export default DeviceRoutes;
