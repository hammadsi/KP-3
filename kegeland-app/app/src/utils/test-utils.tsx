/* istanbul ignore file */
import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {Provider as PaperProvider} from 'react-native-paper';
import {render, RenderOptions} from '@testing-library/react-native';
import {Provider as ReduxProvider} from 'react-redux';

import {store} from '~state/store';
import {lightTheme} from '~constants/theme';
import {initialStore, mockStore} from '~state/ducks/__mocks__/store.mock';

type TestWrapperProps = {
  children: React.ReactNode;
  store?: typeof store;
};

/**
 * Wrapper component with redux state. Used when testing components
 * which requires redux.
 * @param param0 test wrapper props @see {@link TestWrapperProps}
 * @returns
 */
export const TestWrapper: React.FC<TestWrapperProps> = ({
  children,
  ...props
}) => {
  return (
    <ReduxProvider store={props.store || mockStore(initialStore)}>
      <PaperProvider theme={lightTheme}>
        <NavigationContainer theme={lightTheme}>{children}</NavigationContainer>
      </PaperProvider>
    </ReduxProvider>
  );
};

/**
 * Custom test renderer which wraps components with a redux store provider
 * @param ui the component to render
 * @param options renderer options
 * @returns test rendered componnet
 */
export const customRender = <T extends any>(
  ui: React.ReactElement<T>,
  options?: {
    wrapperProps?: {
      store: any;
    };
    renderOptions?: RenderOptions;
  },
) =>
  render(ui, {
    wrapper: (props) => <TestWrapper {...props} {...options?.wrapperProps} />,
    ...options?.renderOptions,
  });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
