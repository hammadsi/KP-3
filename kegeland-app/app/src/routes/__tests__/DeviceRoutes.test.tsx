import React from 'react';

import {render} from '~utils/test-utils';

import DeviceRoutes from '../DeviceRoutes';

describe('Test DeviceRoutes-component', () => {
  it('should render correctly', () => {
    const component = <DeviceRoutes />;
    const tree = render(component);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
