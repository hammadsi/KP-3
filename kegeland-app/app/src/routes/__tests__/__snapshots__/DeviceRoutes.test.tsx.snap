// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DeviceRoutes-component should render correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#f6f6f6",
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <RNSScreenContainer
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RNSScreen
            activityState={2}
            forwardedRef={[Function]}
            gestureResponseDistance={
              Object {
                "bottom": -1,
                "end": -1,
                "start": -1,
                "top": -1,
              }
            }
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "opacity": 1,
                }
              }
            />
            <View
              accessibilityElementsHidden={false}
              closing={false}
              gestureVelocityImpact={0.3}
              importantForAccessibility="auto"
              onClose={[Function]}
              onGestureBegin={[Function]}
              onGestureCanceled={[Function]}
              onGestureEnd={[Function]}
              onOpen={[Function]}
              onTransition={[Function]}
              pointerEvents="box-none"
              style={
                Array [
                  Object {
                    "display": "flex",
                    "overflow": undefined,
                  },
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
              transitionSpec={
                Object {
                  "close": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                  "open": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                }
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                pointerEvents="box-none"
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={false}
                  forwardedRef={[Function]}
                  handlerTag={1}
                  handlerType="PanGestureHandler"
                  needsOffscreenAlphaCompositing={false}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "backgroundColor": "#f6f6f6",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <RNCSafeAreaView
                          edges={
                            Array [
                              "bottom",
                              "left",
                              "right",
                              "top",
                            ]
                          }
                          style={
                            Array [
                              Object {
                                "height": "100%",
                                "marginTop": 8,
                                "width": "100%",
                              },
                              Object {
                                "paddingHorizontal": 16,
                              },
                              undefined,
                            ]
                          }
                          testID="DevicesScreen"
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontWeight": "400",
                                },
                                Object {
                                  "textAlign": "left",
                                },
                                Array [
                                  Object {
                                    "textAlign": "left",
                                  },
                                  Object {
                                    "color": "rgba(0, 0, 0, 0.87)",
                                    "fontFamily": "System",
                                    "fontWeight": "500",
                                    "writingDirection": "ltr",
                                  },
                                  Array [
                                    Object {
                                      "fontSize": 20,
                                      "letterSpacing": 0.15,
                                      "lineHeight": 30,
                                      "marginVertical": 2,
                                    },
                                    Object {
                                      "fontSize": 24,
                                      "fontWeight": "bold",
                                    },
                                  ],
                                ],
                              ]
                            }
                          >
                            Select device
                          </Text>
                          <RCTScrollView
                            style={
                              Object {
                                "marginTop": 16,
                              }
                            }
                          >
                            <View>
                              <View
                                accessibilityState={
                                  Object {
                                    "disabled": true,
                                  }
                                }
                                accessible={true}
                                focusable={false}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Array [
                                    false,
                                    Array [
                                      Object {
                                        "padding": 8,
                                      },
                                      Object {
                                        "backgroundColor": "#fff",
                                        "borderRadius": 15,
                                      },
                                    ],
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "flexDirection": "row",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "rgba(0, 0, 0, 0.54)",
                                          "borderRadius": 24,
                                          "height": 48,
                                          "width": 48,
                                        },
                                        Array [
                                          Object {
                                            "backgroundColor": "rgba(0, 0, 0, 0.54)",
                                          },
                                          Object {
                                            "alignSelf": "center",
                                          },
                                        ],
                                      ]
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "testUri": "../../../assets/devices/femfit.png",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderRadius": 24,
                                          "height": 48,
                                          "width": 48,
                                        }
                                      }
                                    />
                                  </View>
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "marginVertical": 6,
                                          "paddingLeft": 8,
                                        },
                                        Object {
                                          "flex": 1,
                                          "justifyContent": "center",
                                        },
                                      ]
                                    }
                                  >
                                    <Text
                                      numberOfLines={1}
                                      selectable={false}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#000000",
                                            "fontFamily": "System",
                                            "fontWeight": "400",
                                          },
                                          Object {
                                            "textAlign": "left",
                                          },
                                          Array [
                                            Object {
                                              "fontSize": 16,
                                            },
                                            Object {
                                              "color": "rgba(0, 0, 0, 0.87)",
                                            },
                                            undefined,
                                          ],
                                        ]
                                      }
                                    >
                                      Femfit
                                    </Text>
                                    <Text
                                      numberOfLines={2}
                                      selectable={false}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#000000",
                                            "fontFamily": "System",
                                            "fontWeight": "400",
                                          },
                                          Object {
                                            "textAlign": "left",
                                          },
                                          Array [
                                            Object {
                                              "fontSize": 14,
                                            },
                                            Object {
                                              "color": "rgba(0, 0, 0, 0.54)",
                                            },
                                            undefined,
                                          ],
                                        ]
                                      }
                                    >
                                      Not connected
                                    </Text>
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "alignSelf": "center",
                                        "backgroundColor": "rgba(0, 0, 0, 0.12)",
                                        "borderColor": "transparent",
                                        "borderRadius": 15,
                                        "borderStyle": "solid",
                                        "borderWidth": 0,
                                        "elevation": 0,
                                        "minWidth": 64,
                                        "shadowColor": "#000000",
                                        "shadowOffset": Object {
                                          "height": 0,
                                          "width": 0,
                                        },
                                        "shadowOpacity": 0,
                                        "shadowRadius": 0,
                                      }
                                    }
                                  >
                                    <View
                                      accessibilityRole="button"
                                      accessibilityState={
                                        Object {
                                          "disabled": true,
                                        }
                                      }
                                      accessible={true}
                                      focusable={true}
                                      onClick={[Function]}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        Array [
                                          Object {
                                            "overflow": "hidden",
                                          },
                                          Object {
                                            "borderRadius": 15,
                                          },
                                        ]
                                      }
                                    >
                                      <View
                                        style={
                                          Array [
                                            Object {
                                              "alignItems": "center",
                                              "flexDirection": "row",
                                              "justifyContent": "center",
                                            },
                                            undefined,
                                          ]
                                        }
                                      >
                                        <Text
                                          numberOfLines={1}
                                          selectable={false}
                                          style={
                                            Array [
                                              Object {
                                                "color": "#000000",
                                                "fontFamily": "System",
                                                "fontWeight": "400",
                                              },
                                              Object {
                                                "textAlign": "left",
                                              },
                                              Array [
                                                Object {
                                                  "letterSpacing": 1,
                                                  "marginHorizontal": 16,
                                                  "marginVertical": 9,
                                                  "textAlign": "center",
                                                },
                                                undefined,
                                                Object {
                                                  "textTransform": "uppercase",
                                                },
                                                Object {
                                                  "color": "rgba(0, 0, 0, 0.32)",
                                                  "fontFamily": "System",
                                                  "fontWeight": "500",
                                                },
                                                Object {
                                                  "fontFamily": "System",
                                                  "fontWeight": "500",
                                                },
                                                undefined,
                                              ],
                                            ]
                                          }
                                        >
                                          Select
                                        </Text>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </RNCSafeAreaView>
                      </View>
                      <View
                        collapsable={false}
                        nativeID="animatedComponent"
                        pointerEvents="box-none"
                        style={Object {}}
                      >
                        <View
                          accessibilityElementsHidden={false}
                          importantForAccessibility="auto"
                          onLayout={[Function]}
                          pointerEvents="box-none"
                          style={null}
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RNSScreen>
        </RNSScreenContainer>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 44,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 1,
            }
          }
        />
      </View>
    </RNCSafeAreaProvider>
  </View>
</View>
`;
